mmedit/datasets/pipelines/generate_assistant.py

```diff
@@ -167,3 +167,75 @@ def __repr__(self):
         repr_str += (f'sample_quantity={self.sample_quantity}, '
                      f'scale={self.scale}, target_size={self.target_size}')
         return repr_str
+
+
+@PIPELINES.register_module()
+class GenerateLandmark:
+    """Generate face landmark.
+
+    Args:
+        image_key (str): Key of face image.
+        device (str): Device of face alignment. Default: 'cpu'.
+    """
+
+    def __init__(self, image_key, device='cpu'):
+        self.image_key = image_key
+        self.face_alignment_model = None
+        self.init = False
+        self.device = device
+
+    def __call__(self, results):
+        """Call function.
+
+        Args:
+            results (dict): A dict containing the necessary information and
+                data for augmentation. Require image_key.
+
+        Returns:
+            dict: A dict containing the processed data and information.
+                Add 'landmark'.
+        """
+
+        image = results[self.image_key]
+        landmark = self._face_alignment_detector(image)
+        results['landmark'] = landmark
+
+        return results
+
+    def _face_alignment_detector(self, image):
+        """Generate face landmark by face_alignment.
+
+        Args:
+            image (numpy.ndarray): Face image.
+
+        Returns:
+            landmark (Tuple[float]): Location of landmark.
+        """
+
+        try:
+            import face_alignment
+        except ImportError:
+            raise ImportError('please import face-alignment.')
+        if not self.init:
+            self.face_alignment_model = face_alignment.FaceAlignment(
+                face_alignment.LandmarksType._2D,
+                device=self.device,
+                flip_input=False)
+            self.init = True
+        faces = self.face_alignment_model.get_landmarks(image)
+        index = 0
+        max_size = 0
+        for i, face in enumerate(faces):
+            size = face[8, 1] - face[19, 1]
+            if size > max_size:
+                max_size = size
+                index = i
+        landmark = faces[index]
```

Comment:
It seems that this function returns the largest face if multiple faces exist. Better to mention that in the docstrings.